### PR TITLE
feat(config): add environment option

### LINKED ISSUE #88: Add app.environment to payload sent to APM Server
To run Elastic APM on multiple environments of the same service, you'd previously normally add a post-fix to the `serviceName`, e.g: `my-service-staging` or `my-service-prod`.

In Elastic APM 6.2 we'll introduce a new `environment` property tailored specifically for this. This means that users should use the same `serviceName` (`my-service`), and instead specify the environment via a separate property whos value should be sent up to the APM Server using `service.environment` in the payload.

We need to settle on the name of the config option, but we'll probably name it either `env` or `environment`.

Example usage:

```js
require('elastic-apm-node').start({
  serviceName: 'my-service',
  env: 'production'
})
```

The associated environment variable would then be either `ELASTIC_APM_ENV` or `ELASTIC_APM_ENVIRONMENT` depending on what name we end up choosing.

Related APM Server PR: elastic/apm-server#366

### PR DESCRIPTION
**Pull request by Qard**
~~Depends on elastic/apm-nodejs-http-client#58~~
Fixes #88

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation

### CODE REVIEW

File: test/config.js [hunk 1]
```diff
@@ -59,7 +59,8 @@ var optionFixtures = [
   ['kubernetesNamespace', 'KUBERNETES_NAMESPACE'],
   ['kubernetesPodName', 'KUBERNETES_POD_NAME'],
   ['kubernetesPodUID', 'KUBERNETES_POD_UID'],
-  ['usePathAsTransactionName', 'USE_PATH_AS_TRANSACTION_NAME', false]
+  ['usePathAsTransactionName', 'USE_PATH_AS_TRANSACTION_NAME', false],
+  ['environment', 'ENVIRONMENT']
```
Comment (watson) [inline test/config.js]: I think we can add a test to test the default value like so:

```suggestion
  ['environment', 'ENVIRONMENT', 'development']
```

**Comment by Qard** [comment]
Is there a reason _not_ to just land this? Seems like from it being already implemented in the other agents that at least the agent side of things is basically already decided? 🤔

**Comment by watson** [comment]
It was considered a buggy feature as the UI would show invalid data if used. We therefore wanted to pull it from the other agents, but that would have been a breaking change. So instead of opted for a warning in the documentation not to use it for the agents who already released it. Until the UI is released that supports it, it will technically still be problematic if used.

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -91,6 +91,17 @@ var options = {
...
+===== `environment`
+
+* *Type:* String
+* *Default:* `process.env.NODE_ENV || 'development'`
+* *Env:* `ELASTIC_APM_ENVIRONMENT`
+
+The environment name is automatically logged along with all errors and transactions.
+If you want to overwrite this,
+use this option.
+
```
Comment (bmorelli25) [inline docs/configuration.asciidoc]: @Qard Can you add the following as a new paragraph after line 103:
```
Environments allow you to easily filter data on a global level in the APM UI.
It's important to be consistent when naming environments across agents.
See {kibana-ref}/filters.html#environment-selector[environment selector] in the Kibana UI for more information.

NOTE: This feature is fully supported in the APM UI in Kibana versions >= 7.2.
You must use the query bar to filter for a specific environment in versions prior to 7.2.
```

Comment (Qard) [reply docs/configuration.asciidoc]: fixed

**Comment by watson** [comment]
@bmorelli25 The docs-build is still failing. Is that because the new docs have not yet published? If so, how can we merge this before 7.2?

**Comment by bmorelli25** [comment]
@watson unfortunately, the link checker in CI still isn't perfect. I'm not sure why it's failing, but this link does work. I think this PR is ok to merge as is.

**Pull request merged**